Userland/Libraries/LibJS/Bytecode/Pass/UnifySameBlocks.cpp

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (c) 2021, Ali Mohammad Pur <mpfard@serenityos.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <LibJS/Bytecode/PassManager.h>
+#include <string.h>
+
+namespace JS::Bytecode::Passes {
+
+void UnifySameBlocks::perform(PassPipelineExecutable& executable)
+{
+    started();
+
+    VERIFY(executable.cfg.has_value());
+    VERIFY(executable.inverted_cfg.has_value());
+    auto cfg = executable.cfg.release_value();
+    auto inverted_cfg = executable.inverted_cfg.release_value();
+
+    HashMap<BasicBlock const*, BasicBlock const*> equal_blocks;
+
+    for (size_t i = 0; i < executable.executable.basic_blocks.size(); ++i) {
+        auto& block = executable.executable.basic_blocks[i];
+        auto block_bytes = block->instruction_stream();
+        for (auto& candidate_block : executable.executable.basic_blocks.span().slice(i + 1)) {
+            if (equal_blocks.contains(&*candidate_block))
+                continue;
+            // FIXME: This can probably be relaxed a bit...
+            if (candidate_block->size() != block->size())
+                continue;
+
+            auto candidate_bytes = candidate_block->instruction_stream();
+            // FIXME: NewBigInt's value is not correctly reflected by its encoding in memory,
+            //        this will yield false negatives for blocks containing that
+            if (memcmp(candidate_bytes.data(), block_bytes.data(), candidate_block->size()) == 0)
+                equal_blocks.set(candidate_block.ptr(), block);
+        }
```

Comment:
Not sure how effective this is with the new Operands, which may not be deduplicated

Reply:
On SunSpider and Kraken, it's pretty ineffective. Only a couple blocks get unified. It gets going on Octane though, where it finds hundreds of good candidates.